### PR TITLE
Implement filter tree editor

### PR DESCRIPTION
**Pull request by julianbreiteneicher**
Add a graphical editor to create/edit commit filters that can be used by VaRA's analysis.
The created filters are stored in a yaml file and can be loaded by VaRA.

### CODE REVIEW

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 18
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

**Review by vulder — APPROVED** [review]
lgtm

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 3
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

File: varats/data/filtertree_data.py [hunk 1]
```diff
@@ -0,0 +1,819 @@
+import yaml
+
+from typing import List
+from PyQt5.QtCore import QDateTime, Qt
+
+
+"""
```
Comment (vulder) [inline varats/data/filtertree_data.py:1]: Move the doc at the top of the file so it is recognized as module doc, imports come afterwards.

Comment (julianbreiteneicher) [reply varats/data/filtertree_data.py:1]: Ah, ok :)

**Comment by vulder** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Clones added
============
- varats/driver.py  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/app/VaRA/VaRA-Tool-Suite/pullRequest?prid=3503365&bid=12493914)

File: varats/data/filtertree_data.py (hunk 1 above)
Comment (julianbreiteneicher) [reply varats/data/filtertree_data.py:1]: Done

**Review by vulder — APPROVED** [review]
lgtm

**Pull request merged**